pkg/products/monitoringspec/reconciler.go

```diff
@@ -0,0 +1,388 @@
+			if err != nil && k8serr.IsNotFound(err) {
+				logrus.Infof("Spec phase completed")
+				//namespace is gone, return complete
+				return integreatlyv1alpha1.PhaseCompleted, nil
+			}
+			phase, err := resources.RemoveNamespace(ctx, installation, serverClient, r.Config.GetNamespace())
+			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+				logrus.Infof("Spec phase removal failure: %v", err)
+				return phase, err
+			}
+			return integreatlyv1alpha1.PhaseInProgress, nil
+		})
+
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		logrus.Errorf("failed to reconcile finalizer: %v", err)
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile finalizer", err)
+		return phase, err
+	}
+
+	phase, err = r.createNamespace(ctx, serverClient, installation)
+	logrus.Infof("Phase: %s createNamespace", phase)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		logrus.Errorf("failed to create namespace : %v", err)
+		events.HandleError(r.recorder, installation, phase, "Failed to create namespace", err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileMonitoring(ctx, serverClient, installation)
+	logrus.Infof("Phase: %s reconcileMonitoring", phase)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		logrus.Errorf("ailed to reconcile: %v", err)
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile:", err)
+		return phase, err
+	}
+
+	product.Host = r.Config.GetHost()
+	product.Version = r.Config.GetProductVersion()
+
+	err = r.ConfigManager.WriteConfig(r.Config)
+	if err != nil {
+		logrus.Errorf("failed to write config: %v ", err)
+		events.HandleError(r.recorder, installation, integreatlyv1alpha1.PhaseFailed, "Failed to update monitoring config", err)
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("could not update monitoring config: %w", err)
+	}
+
+	events.HandleProductComplete(r.recorder, installation, integreatlyv1alpha1.MonitoringStage, r.Config.GetProductName())
+	logrus.Infof("%s installation is reconciled successfully", packageName)
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+// make the federation namespace discoverable by cluster monitoring
+func (r *Reconciler) createNamespace(ctx context.Context, serverClient k8sclient.Client,
+	installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	namespace, err := resources.GetNS(ctx, r.Config.GetNamespace(), serverClient)
+	if err != nil {
+		if !k8serr.IsNotFound(err) {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+		_, err := resources.CreateNSWithProjectRequest(ctx, r.Config.GetNamespace(),
+			serverClient, installation, false, true)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+		return integreatlyv1alpha1.PhaseCompleted, nil
+	}
+
+	//TODO - Cluster monitoring flipped back to true
+	resources.PrepareObject(namespace, installation, false, false)
```

Comment:
@alanmoran We can set the last arg to `true` here once we're happy to add the `openshift.io/cluster-monitoring=true` label & start scraping product metrics into the cluster prometheus

Reply:
Perfect!

Reply:
@Rajagopalan-Ranganathan Can you set the value to true.
We are good to go on scraping these metrics.

Reply:
Set the cluster monitoring label to true.